python/cudf/cudf/tests/test_indexing.py

```diff
@@ -996,7 +1006,7 @@ def test_sliced_indexing():
     pidx = pdf.index[:75]
     gidx = gdf.index[:75]
 
-    assert_eq(pdf.loc[pidx], gdf.loc[gidx])
+    assert_eq(pdf.loc[pidx].sort_index(), gdf.loc[gidx].sort_index())
```

Comment:
We shouldn't need to sort by the index in the case of `.loc`. It should have a deterministic and consistent output ordering.

Reply:
But don't think as we discussed join would not yield a ordered output. we might have to again sort the dataframe as per older order.

Reply:
Yea we need to sort it based on the input order within the `.loc` operation.

Reply:
Sorting with one more column attached to other_df.